src/pudl/io_managers.py

```diff
@@ -83,6 +87,72 @@ def __getitem__(self, idx):
         return self.fk_errors[idx]
 
 
+def get_table_name_from_context(context: OutputContext) -> str:
+    """Retrieves the table name from the context object."""
+    # TODO(rousik): Figure out which kind of identifier is used when.
+    if context.has_asset_key:
+        return context.asset_key.to_python_identifier()
+    return context.get_identifier()
+
+
+class PudlMixedFormatIOManager(ConfigurableIOManager):
+    """Format switching IOManager that supports sqlite and parquet.
+
+    We currently support both sqlite and parquet formats. This io manager
+    allows us to choose which formats should be used for reading and which
+    should be used for reading.
+
+    This should allow us to develop support for parquet format without
+    affecting the existing sqlite-based functionality.
+
+    Switching between reads from sqlite and parquet should allow us
+    to test the two formats for eqivalence during development.
+    """
+
+    # TODO(rousik): the folowing EnvVars should really be booleans but
+    # dagster EnvVar doesn't recognize anything but str/int types here
+    # and AFAIK there's no way to set default value if the env variable
+    # is not set.
+
+    write_to_parquet: bool = bool(EnvVar("PUDL_WRITE_TO_PARQUET").get_value(False))
+    """If true, data will be written to parquet files."""
+
+    read_from_parquet: bool = bool(EnvVar("PUDL_READ_FROM_PARQUET").get_value(False))
+    """If true, data will be read from parquet files instead of sqlite."""
+
+    def __init__(self, **kwargs):
+        """Creates new instace of the IO Manager."""
+        super().__init__(**kwargs)
+        # TODO(rousik): Init arguments are fixed but we do not expect
+        # any variability here. If we do, we should add this to the
+        # configuration.
+        self._sqlite_io_manager = PudlSQLiteIOManager(
+            base_dir=PudlPaths().output_dir,
+            db_name="pudl",
+        )
+        self._parquet_io_manager = PudlParquetIOManager()
+
+        # TODO(rousik): Move the following diagnostics to debug level once
+        # we know what is happening.
+        write_fmt = "parquet" if self.write_to_parquet else "sqlite"
+        read_fmt = "parquet" if self.read_from_parquet else "sqlite"
+        logger.warning(
+            f"pudl_io_manager will be reading from {read_fmt}, writing to {write_fmt}."
+        )
```

Comment:
There's an asymmetry here in that we can only read from one source, and it can be either SQLite or Parquet, but we're always writing to SQLite, and optionally also writing to Parquet. But the way it's phrased it seems like there's one read format, and one write format, which is misleading.

Reply:
Right, I have tweaked the log message. We probably want to drop the severity also. I have used `warning` for local testing to see that it correctly picks up env var overrides.